src/database/migration/file/migration_version_7.ts

```diff
@@ -0,0 +1,45 @@
+import {SQLiteDatabase} from 'react-native-sqlite-storage';
+
+import ChannelList from '../../schema/ChannelListSchema';
+import ChatSchema from '../../schema/ChatSchema';
+import Migration from './migration.types';
+
+const TABLE_NAME = ChatSchema.getTableName();
+
+class MigrationVersion7 implements Migration {
+  up = async (db: SQLiteDatabase): Promise<void> => {
+    const upQueryStep1 = `CREATE TABLE IF NOT EXISTS ${TABLE_NAME}_temp (
+        id UUID PRIMARY KEY NOT NULL,
+        channel_id TEXT NOT NULL,
+        user_id UUID NOT NULL,
+        message TEXT NOT NULL,
+        type TEXT NOT NULL,
+        status TEXT NOT NULL,
+        created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+        updated_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+        raw_json TEXT,
+        FOREIGN KEY (channel_id) REFERENCES ${ChannelList.getTableName()} (id),
+        FOREIGN KEY (user_id) REFERENCES users (id)
+      );`;
+
+    const upQueryStep2 = `INSERT INTO ${TABLE_NAME}_temp SELECT * FROM ${TABLE_NAME};`;
+    const upQueryStep3 = `DROP TABLE ${TABLE_NAME};`;
+    const upQueryStep4 = `ALTER TABLE ${TABLE_NAME}_temp RENAME TO ${TABLE_NAME};`;
+
+    console.log(`===== MIGRATING: ${TABLE_NAME.toLocaleUpperCase()} TABLE =====`);
+    await db.executeSql(upQueryStep1);
+    await db.executeSql(upQueryStep2);
+    await db.executeSql(upQueryStep3);
+    await db.executeSql(upQueryStep4);
+    console.log(`===== DONE MIGRATING: ${TABLE_NAME.toLocaleUpperCase()}  TABLE =====`);
+  };
+
+  down = async (db: SQLiteDatabase): Promise<void> => {
+    console.log(`===== DOWNGRADING: ${TABLE_NAME.toLocaleUpperCase()} TABLE =====`);
+    const downQuery = `DROP TABLE IF EXISTS ${TABLE_NAME}`;
+    await db.executeSql(downQuery);
+    console.log(`===== DONE DOWNGRADING: ${TABLE_NAME.toLocaleUpperCase()}  TABLE =====`);
+  };
+}
+
+export default new MigrationVersion7();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The migration script seems to be well written and follows the standard practice of creating a temporary table, copying data from the old table to the new one, dropping the old table, and renaming the temporary table. However, it's important to note that this operation is not atomic and can lead to data loss if an error occurs during the process. 

To improve the robustness of your migration, you could wrap the operations in a transaction. This way, if any step fails, the database will roll back to its previous state, preventing potential data loss.

Here's how you could modify your code:

```diff
+    db.transaction(async tx => {
+      console.log(`===== MIGRATING: ${TABLE_NAME.toLocaleUpperCase()} TABLE =====`);
+      await tx.executeSql(upQueryStep1);
+      await tx.executeSql(upQueryStep2);
+      await tx.executeSql(upQueryStep3);
+      await tx.executeSql(upQueryStep4);
+      console.log(`===== DONE MIGRATING: ${TABLE_NAME.toLocaleUpperCase()}  TABLE =====`);
+    });
```

Also, consider removing the `console.log` statements or replacing them with a proper logging mechanism before deploying to production. Console logs can slow down your application and expose potentially sensitive information.


<!-- This is an auto-generated comment by OSS CodeRabbit -->